packages/runtime/datastore-hot-swap/src/hotSwapDataStoreRuntime.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { FluidObject } from "@fluidframework/core-interfaces";
+import {
+	IChannel,
+	IChannelFactory,
+	IFluidDataStoreRuntime,
+} from "@fluidframework/datastore-definitions";
+import { assert } from "@fluidframework/core-utils";
+import { FluidDataStoreRuntime, ISharedObjectRegistry } from "@fluidframework/datastore";
+import { ILocalChannelContext, IModifiableFluidDataStoreContext } from "./types";
+
+/**
+ * DO NOT USE THIS CLASS DIRECTLY.
+ */
+export class UnsafeHotSwapFluidDataStoreRuntime extends FluidDataStoreRuntime {
+	private readonly replacedContexts: Map<string, ILocalChannelContext> = new Map();
+	public constructor(
+		private readonly modifiableDataStoreContext: IModifiableFluidDataStoreContext,
+		private readonly _sharedObjectRegistry: ISharedObjectRegistry,
+		existing: boolean,
+		initializeEntryPoint?: (runtime: IFluidDataStoreRuntime) => Promise<FluidObject>,
+	) {
+		super(modifiableDataStoreContext, _sharedObjectRegistry, existing, initializeEntryPoint);
+	}
+
+	// Returns a detached channel
+	public [".UNSAFE_replaceChannel"](id: string, channelFactory: IChannelFactory): IChannel {
```

Comment:
This API should change once we have the actual migrate/barrier op. From that we can then do the ChannelFactory switch. For now, I expose the `channelFactory` for testing purposes.